### ISSUE TITLE
Folder icons show up as missing when using certain globbing patterns

### ISSUE THREAD

**Issue by bording** [dotnet/project-system#2134]
I'm not entirely sure where this bug is best filed, but since I'm using the new project system to reproduce it, I figured I'd start here.

I've got a NuGet package that brings in source files via the contentFiles mechanism. I need to be able to exclude certain files, so I added a `Compile` item with a wildcard pattern to remove them. That works correctly, however it seems to also remove files that aren't part of the pattern as well.

I've pushed up a repo that demonstrates the problem here: https://github.com/bording/GlobbingProblem

If you look at the project file, you can see the following:

```
 <ItemGroup>
    <Compile Remove="$(NuGetPackageRoot)**\Audit\*.cs" />
  </ItemGroup>
```

That removes the 5 files in the Audit folder as expected, but it also removes `ConfigureEndpointLearningTransport.cs`, which doesn't match the pattern at all.

Before remove:
![before](https://cloud.githubusercontent.com/assets/753669/25767110/c2ca36f8-31c4-11e7-96ae-57e4cf920ddd.png)

After remove:
![after](https://cloud.githubusercontent.com/assets/753669/25767113/c72633b4-31c4-11e7-8113-8e3056baaa79.png)

I've experimented with different wildcard patterns, and I can't see any discernible pattern to explain the behavior.

For example:
```
  <ItemGroup>
    <Compile Remove="$(NuGetPackageRoot)**\EndpointTemplates\*.cs" />
  </ItemGroup>
```
Removes files from the Pipeline and Recoverability folders as well.

**Comment by davkean** [comment]
This issue was moved to Microsoft/msbuild#2069

**Comment by davkean** [comment]
MSBuild interprets the wild card, so I've moved it there to start. However, I'm going to leave this bug open because the folder icons are wrong.

**Comment by bording** [comment]
@davkean Thanks for routing it to the correct place. Another related problem with `contentFiles` is that those files are editable even though they are linked from the NuGet package cache and should be immutable.

**Comment by davkean** [comment]
@bording It's known about that content items from NuGet package cache.

The globbing issue has been fixed in 15.3, and also I cannot longer repro the missing folder icon issue. Looks like this was resolved as part of getting linked items showing up by default.
